test/tests/functional/pbs_reservations.py

```diff
@@ -2235,3 +2235,30 @@ def test_server_recover_resv_queue(self):
         jid = self.server.submit(J)
 
         self.server.expect(JOB, {ATTR_state: 'R'}, id=jid)
+
+    @skipOnCpuSet
+    def test_resv_job_hard_walltime(self):
+        """
+        Test that a job with hard walltime will not conflict with
+        reservtion if hard walltime is less that reservation start time.
+        """
+        a = {'resources_available.ncpus': 4}
+        self.server.manager(MGR_CMD_SET, NODE, a, id=self.mom.shortname)
+
+        now = int(time.time())
+
+        a = {'Resource_List.ncpus': 4, 'reserve_start': now + 65,
+             'reserve_end': now + 240}
+        R = Reservation(TEST_USER, attrs=a)
+        rid = self.server.submit(R)
+        self.server.expect(RESV,
+                           {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')},
+                           id=rid)
+
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'False'})
+        a = {'Resource_List.ncpus': 4,
+             'Resource_List.walltime': 60}
```

Comment:
I'd decrease this.  Right now there is a race condition that if it takes more than 5 seconds to submit the reservation, do the expect(), and then submit the job, the job won't run.  We have some super-slow machines in the testbed where it can take more than 5s to do a pbs_rstat.  I'd decrease the walltime to 50s.